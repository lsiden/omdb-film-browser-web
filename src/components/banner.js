import React from "react"
import QueryForm from "query-form"
import { OMDB_URL } from "constants"

const headerStyle = {
  background: "linear-gradient(to right, #000040, #3b00ff)",
  padding: "10px",
  color: "white",
}
const titleStyle = {
  fontSize: "18pt",
  marginBottom: "18pt",
}

const linkStyle = {
  color: "white",
  fontSize: "small",
}

const Banner = () => (
  <header style={headerStyle}>
    <h1 style={titleStyle}>
      {"Search Open Movie Database"}&nbsp;
      <a href={OMDB_URL} style={linkStyle}>
        (OMDB)
      </a>
    </h1>
    <QueryForm />
  </header>
)

export default Banner
