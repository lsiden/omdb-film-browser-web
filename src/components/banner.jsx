import React from 'react'

import { headerStyle, titleStyle } from 'style'
import { BANNER_TITLE } from 'omdb_constants'

const AboutLink = () => (
  <a href="/about" style={{ color: 'white' }}>
    about
  </a>
)

const inAboutPath = () => {
  return /about\/?$/.test(window.location.href)
}

export default () => {
  return (
    <header style={headerStyle}>
      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <h1 style={titleStyle}>{BANNER_TITLE}</h1>
        {!inAboutPath() && <AboutLink />}
      </div>
    </header>
  )
}