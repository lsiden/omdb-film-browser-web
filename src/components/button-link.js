import React from "react"
import PropTypes from "prop-types"

const ButtonLink = (props) => <button className="btn btn-link" {...props} />

ButtonLink.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
    .isRequired,
}

export default ButtonLink
