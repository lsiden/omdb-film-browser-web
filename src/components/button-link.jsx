import React from 'react'
import PropTypes from 'prop-types'

const ButtonLink = ({ style, onClick, children }) => (
  /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
  <button
    type="button"
    className="btn btn-link"
    style={style}
  >
    {children}
  </button>
)

ButtonLink.propTypes = {
  style: PropTypes.object,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
}

ButtonLink.defaultProps = {
  style: {},
}

export default ButtonLink
