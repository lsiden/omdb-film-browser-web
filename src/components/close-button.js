import React from "react"
import { Link } from "react-router-dom"

import ButtonLink from "components/button-link"

const CLOSE_ICON = "\u2715"

const closeButtonStyle = {
  margin: "8pt",
  fontSize: "20pt",
}

export const CloseButton = () => (
  <Link to="/">
    <ButtonLink style={closeButtonStyle} title="Close">
      {CLOSE_ICON}
    </ButtonLink>
  </Link>
)

export default CloseButton
