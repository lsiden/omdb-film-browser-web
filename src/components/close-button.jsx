import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import ButtonLink from 'components/button-link'

const CLOSE_ICON = '\u2715'

const closeButtonStyle = {
  margin: 20,
  fontSize: 20,
}

const CloseButton = ({ buttonStyle, onClick }) => {
  const style = {
    ...closeButtonStyle,
    ...buttonStyle,
  }
  return (
    <Link to="/">
      <ButtonLink onClick={onClick} style={style} title="Close">
        {CLOSE_ICON}
      </ButtonLink>
    </Link>
  )
}
CloseButton.propTypes = {
  buttonStyle: PropTypes.object,
}
CloseButton.defaultProps = {
  buttonStyle: {},
}

export default CloseButton
