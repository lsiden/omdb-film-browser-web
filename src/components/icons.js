import React from "react"

const SEARCH_ICON = "\u26b2"
const CANCEL_ICON = "\u274c"
const ICON_COLOR = "#888"

const searchIconStyle = {
  position: "absolute",
  top: "3pt",
  left: "6pt",
  color: ICON_COLOR,
  fontSize: "20pt",
  transform: "rotate(-45deg)",
}

export const SearchIcon = () => (
  <span
    style={searchIconStyle}
    charSet="utf-8"
    dangerouslySetInnerHTML={{ __html: SEARCH_ICON }}
  />
)

const cancelIconStyle = {
  position: "absolute",
  top: 5,
  right: 5,
  fontSize: "18pt",
  color: ICON_COLOR,
}

export const CancelIcon = (props = {}) => (
  <span
    className="hover-pointer"
    style={cancelIconStyle}
    charSet="utf-8"
    dangerouslySetInnerHTML={{ __html: CANCEL_ICON }}
    {...props}
  />
)
