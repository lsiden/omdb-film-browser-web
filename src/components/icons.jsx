import React from 'react'
import PropTypes from 'prop-types'

const SEARCH_ICON = '\u26b2'
const CANCEL_ICON = '\u274c'
const ICON_COLOR = '#888'

const searchIconStyle = {
  position: 'absolute',
  top: '3pt',
  left: '6pt',
  color: ICON_COLOR,
  fontSize: '20pt',
  transform: 'rotate(-45deg)',
}

export const SearchIcon = () => (
  <span
    style={searchIconStyle}
    charSet="utf-8"
    dangerouslySetInnerHTML={{ __html: SEARCH_ICON }}
  />
)

const cancelIconStyle = {
  position: 'absolute',
  top: 5,
  right: 5,
  fontSize: '18pt',
  color: ICON_COLOR,
}

export const CancelIcon = ({ onClick }) => (
  <span
    role="button"
    tabIndex={0}
    aria-label="Cancel"
    className="hover-pointer"
    style={cancelIconStyle}
    charSet="utf-8"
    dangerouslySetInnerHTML={{ __html: CANCEL_ICON }}
    onClick={onClick}
    onKeyPress={onClick}
  />
)

CancelIcon.propTypes = {
  onClick: PropTypes.func,
}

CancelIcon.defaultProps = {
  onClick: () => {},
}
