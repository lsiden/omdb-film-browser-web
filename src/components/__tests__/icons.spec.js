import React from "react"
import { shallow } from "enzyme"

import { CancelIcon, SearchIcon } from "components/icons"

test("SearchIcon matches snapshot", () => {
  expect(shallow(<SearchIcon />)).toMatchSnapshot()
})

test("CancelIcon matches snapshot", () => {
  expect(shallow(<CancelIcon />)).toMatchSnapshot()
})
