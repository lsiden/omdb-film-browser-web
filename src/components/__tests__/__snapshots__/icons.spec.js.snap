// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelIcon matches snapshot 1`] = `ShallowWrapper {}`;

exports[`SearchIcon matches snapshot 1`] = `ShallowWrapper {}`;
